src/vs/workbench/contrib/chat/browser/chatListRenderer.ts

```diff
@@ -367,9 +370,23 @@ export class ChatListItemRenderer extends Disposable implements ITreeRenderer<Ch
 	}
 
 	private renderDetail(element: IChatResponseViewModel, templateData: IChatListItemTemplate): void {
-		let progressMsg: string = '';
+		let agentName: IObservable<string | undefined> = constObservable(undefined);
+
 		if (element.agent && !element.agent.isDefault) {
-			let usingMsg = chatAgentLeader + element.agent.name;
+			const name = element.agent.name;
+			agentName = this.chatAgentNameService.getAgentNameRestriction(element.agent)
+				.map(allowed => allowed ? name : name); // TODO
```

Comment:
@roblourens This would be the place to render it differently, based on whether it's allowed.